services/item.py

```diff
@@ -67,10 +68,12 @@ async def get_items_by_buy_id(buy_id: int) -> list:
 
     @staticmethod
     async def get_unsold_subcategories_by_category(category_id: int, page) -> list[Item]:
-        async with async_session_maker() as session:
-            stmt = select(Item).join(Subcategory, Subcategory.id == Item.subcategory_id).where(
+        async with (async_session_maker() as session):
+            stmt = (select(Item, Category.name, Subcategory.name)
+                    .join(Subcategory, Subcategory.id == Item.subcategory_id)
+                    .join(Category, Category.id == category_id).where(
                 Item.category_id == category_id, Item.is_sold == 0).group_by(Subcategory.name).limit(
-                ItemService.items_per_page).offset(ItemService.items_per_page * page)
+                ItemService.items_per_page).offset(ItemService.items_per_page * page))
```

Comment:
Why did you redo the query if in create_subcategory_buttons, item.subcategory.name and item.category.name are not used?
As far as I know joinedload is disabled here.